openedx/core/djangoapps/user_api/api.py

```diff
@@ -415,14 +415,23 @@ def _add_password_field(self, form_desc, required=True):
         # meant to hold the user's password.
         password_label = _(u"Password")
 
+        restrictions = {
+            "min_length": password_min_length(),
+            "max_length": password_max_length(),
+        }
+
+        complexities = getattr(settings, 'PASSWORD_COMPLEXITY', {})
+        if not settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
+            complexities = {}
+        for key, value in complexities.iteritems():
+            api_key = key.lower().replace(' ', '_')
+            restrictions[api_key] = value
```

Comment:
Just pointing out that min/max length could be overwritten via `complexities`. Is that ok?

Reply:
Yeah, you're right. I figured ever-so-slightly cleaner code (rather than adding min/max to the dictionary later separately from dictionary declaration) was worth the risk that some admin would encode length in the wrong place.

We don't have any reporting currently for if a site admin uses invalid values in the complexities field. :man_shrugging:

Reply:
If it's not a 'feature' to be able to override length, I'd prefer to add it after the complexities are added.

Reply:
Alright. I don't necessarily agree on the tradeoff, but no biggie. Updated to be maximally correct.

Reply:
I think my preferred fix would be to have a utility function somewhere like get_password_complexities() that returns a validated list so we could trust the values aren't whack. I didn't do it as part of this work, but next time I'm in there I'll see if it makes sense to add.